### PR TITLE
Add Javascript support

### PR DESCRIPTION
**Pull request by jheysel-r7**
Small update to add javascript as a supported language when generating random identifiers.

### CODE REVIEW

File: lib/rex/random_identifier/generator.rb [hunk 1]
```diff
@@ -62,10 +62,42 @@ class ExhaustedSpaceError < StandardError; end
     ).uniq.freeze
   )
 
+  JavaScriptOpts = JavaOpts.merge(
+    forbidden: (
+      JavaOpts[:forbidden] +
+        # JavaScript Reserved Words + JavaScript Objects, Properties, and Methods + Java Reserved Words + Other Reserved Words + HTML Event Handlers
+        # https://www.w3schools.com/js/js_reserved.asp
```
Comment (adfoster-r7) [inline lib/rex/random_identifier/generator.rb:69]: Looks like your list is misisng keywords like `const`, `class` etc; it's also missing identifiers with special meanings - such as `async`. I'd probably recommend MDN:

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#reserved_words
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#identifiers_with_special_meanings
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#future_reserved_words

Or the specification:

https://tc39.es/ecma262/#sec-keywords-and-reserved-words

File: lib/rex/random_identifier/generator.rb [hunk 2]
```diff
@@ -62,10 +62,42 @@ class ExhaustedSpaceError < StandardError; end
...
+          event fileUpload focus form forms frame innerHeight innerWidth
+          layer layers link location mimeTypes navigate navigator frames
+          frameRate hidden history image images offscreenBuffering open opener
+          option outerHeight outerWidth packages pageXOffset pageYOffset parent parseFloat
+          parseInt password pkcs11 plugin  prompt propertyIsEnum radio reset
+          screenX screenY scroll secure select self setInterval setTimeout
+          status submit taint text  textarea top unescape untaint window
+
+          onblur onclick onerror onfocus onkeydown onkeypress onkeyup onmouseover
+          onload onmouseup onmousedown onsubmit
```
Comment (adfoster-r7) [inline lib/rex/random_identifier/generator.rb:91]: We should be good to skip these, they're valid javascript identifiers for both node and javascript running in the browser

```suggestion
```

```
$ node
Welcome to Node.js v22.6.0.
Type ".help" for more information.
> var onblur = 'hello';
undefined
> console.log(onblur)
hello
undefined
```

File: lib/rex/random_identifier/generator.rb (hunk 1 above)
Comment (jheysel-r7) [reply lib/rex/random_identifier/generator.rb]: I wasn't sure if this should be an explicit list or not - I had removed `const`, `class` and all those already defined in `JavaOpts[:forbidden]` as it adds those in with:
```
    forbidden: (
      JavaOpts[:forbidden] +
```

 I've added them back in and updated the list to comply with this words listed in the URLs above. Thank you!

Comment (adfoster-r7) [reply lib/rex/random_identifier/generator.rb]:
```
    forbidden: (
      JavaOpts[:forbidden] +
```

Ah I didn't spot that on the first read, I Wouldn't have expected JavaScript's word list to extend Java's (https://en.wikipedia.org/wiki/Principle_of_least_astonishment 😄) but that's all good - apologies :+1:

Comment (adfoster-r7) [reply lib/rex/random_identifier/generator.rb]: I think this updated list is still missing the inbuilt window/global instance methods like btoa etc:

https://github.com/rapid7/rex-random_identifier/pull/12#discussion_r1840804575

Which folk may be reaching for in their payloads

```
$n ode
Welcome to Node.js v22.6.0.
Type ".help" for more information.
> btoa
[Function: btoa]
```

**Review by smcintyre-r7 — APPROVED** [review]
Tested as part of rapid7/metasploit-framework#19640.

**Pull request merged**